examples/advanced/vertical_xgboost/utils/prepare_data.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import shutil
+
+import numpy as np
+import pandas as pd
+
+
+def data_split_args_parser():
+    parser = argparse.ArgumentParser(description="Generate data split for dataset")
+    parser.add_argument("--data_path", type=str, help="Path to data file")
+    parser.add_argument("--site_num", type=int, help="Total number of sites")
+    parser.add_argument("--site_name_prefix", type=str, default="site-", help="Site name prefix")
+    parser.add_argument("--cols_total", type=int, help="Total number of columns")
+    parser.add_argument("--rows_overlap", type=int, help="Size of the data overlap between sites")
+    parser.add_argument("--rows_total", type=int, help="Total number of instances")
+    parser.add_argument("--out_path", type=str, default="~/dataset", help="Output path for the data split json file")
+    return parser
+
+
+def split_num_proportion(n, site_num):
+    split = []
+    ratio_vec = np.ones(site_num)
+    total = sum(ratio_vec)
+    left = n
+    for site in range(site_num - 1):
+        x = int(n * ratio_vec[site] / total)
+        left = left - x
+        split.append(x)
+    split.append(left)
+    return split
+
+
+def main():
+    parser = data_split_args_parser()
+    args = parser.parse_args()
+    print(f"site_num: {args.site_num}")
+    print(f"rows_total: {args.rows_total}")
+    print(f"rows_overlap: {args.rows_overlap}")
+    print(f"cols_total: {args.cols_total}")
+
```

Comment:
Sure I agree with getting cols_total from the dataframe, however psi takes an unreasonable amount of time (>30min) when using all 11,000,000 rows, so that's why I put an option to use less than that